src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +392,460 @@ else if (!period.isEmpty()) {
+                        getStringInsideHtmlElement(child.getChildren()[0]);
+                result.append(stringInsideHtmlTags);
+                final boolean isHtmlList = HTML_LIST_TAGS.contains(getHtmlTagName(child));
+                if (isHtmlList) {
+                    for (DetailNode nodeChild : child.getChildren()[0].getChildren()) {
+                        if (nodeChild.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+                            result.append(nodeChild.getChildren()[0].getChildren()[1].getText());
+                        }
+                    }
+                }
+            }
+        }
+        return result.toString();
+    }
+
+    /**
+     * Get html tag name.
+     *
+     * @param htmlNode html element node.
+     * @return Name of html tag.
+     */
+    private static String getHtmlTagName(DetailNode htmlNode) {
+        // This will never return NPE as htmlNode will at have least one child
+        // This child will have at least 3 children
+        // The first child is of type HTML_ELEMENT_START which always have at least 2 elements.
+        return htmlNode.getChildren()[0].getChildren()[0].getChildren()[1].getText();
+    }
+
+    /**
+     * Get text inside html elements.
+     *
+     * @param htmlNode html element node.
+     * @return text inside html element.
+     */
+    private static String getStringInsideHtmlElement(DetailNode htmlNode) {
+        final StringBuilder htmlString = new StringBuilder(250);
+        // This will never return NPE as htmlNode will at have least one child
+        // This child will have at least 3 children
+        for (DetailNode child: htmlNode.getChildren()) {
+            if (child.getType() == JavadocTokenTypes.TEXT) {
+                htmlString.append(child.getText());
+            }
+        }
+        return htmlString.toString();
+    }
+
+    /**
+     * Gets summary sentence for html inline type summary.
+     * Html type summary sentence are wrapped within paragraph tag.
+     *
+     * @param ast Javadoc root node.
+     * @return first sentence.
+     */
+    private static String getInlineHtmlSentence(DetailNode ast) {
+        final StringBuilder result = new StringBuilder(256);
+        final DetailNode inlineTagNode = getInlineTagNodeInsideHtmlTagsOnly(ast);
+        result.append(getFirstSentenceFromInlineTagNode(
+                Objects.requireNonNull(inlineTagNode)));
+        return result.toString();
+    }
+
+    /**
+     * Finds and returns description node.
+     *
+     * @param ast Javadoc root node.
+     * @return Description DetailNode.
+     */
+    private static DetailNode getDescriptionNode(DetailNode ast) {
+        DetailNode descriptionNode = null;
+        for (DetailNode node : ast.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                for (DetailNode child : node.getChildren()) {
+                    if (child.getType() == JavadocTokenTypes.DESCRIPTION) {
```

Comment:
please reuse https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtil.java#L201

Reply:
Done. Changed here and at 3 other methods.